### PR TITLE
Handle errors

### LINKED ISSUE #15: Handle errors
- [x] Printing `errno` on error would be very useful to debug socket communications.
- [x] Implement an error message for cases when get_monitors fails (WM doesn't support _NET_DESKTOP_VEWPORT)
- [x] Error handling in daemon
- [x] Error handling in dexpo
- [x] Handle xcb errors with `xcb_generic_error_t`

### LINKED ISSUE #32: Move `window` related functions from `dexpo.cpp` to `window.cpp`
This would simplify error handling

### PR DESCRIPTION
**Pull request by mmskv**
Closes #15 
- [x] Printing `errno` on error would be very useful to debug socket communications.
- [x] Implement an error message for cases when get_monitors fails (WM doesn't support _NET_DESKTOP_VEWPORT)
- [x] Error handling in daemon
- [x] Error handling in dexpo
- [x] Handle xcb errors with `xcb_generic_error_t`

### CODE REVIEW

File: src/config.hpp [hunk 1]
```diff
@@ -3,26 +3,25 @@
...
 const uint16_t dexpo_padding = 5; ///< Padding around the screenshots
 
 const uint16_t dexpo_hlwidth = 3; ///< Width of the preselected desktop border
 const uint32_t dexpo_hlcolor = 0XFF662C28; ///< Color of the preselection border
 const uint32_t dexpo_bgcolor = 0xAA525252; ///< Background color. TODO Alpha
 
-const int16_t dexpo_x = 0; ///< X coordinate of window's top left corner
-const int16_t dexpo_y = 0; ///< Y coordinate of window's top left corner
+const int16_t dexpo_x = 1080; ///< X coordinate of window's top left corner
+const int16_t dexpo_y = 20;   ///< Y coordinate of window's top left corner
```
Comment (TheHussky) [inline src/config.hpp]: Why is it 1080 and 20?

File: src/dexpo-daemon.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+main ()
+{
+  try
+    {
+      auto d = dxp_daemon ();
+      d.run ();
+    }
+  catch (const std::runtime_error &e)
+    {
+      std::cerr << e.what () << std::endl;
```
Comment (TheHussky) [inline src/dexpo-daemon.cpp:23]: unusual naming. Did You name it yourself? may I suggest e.type() or e.info() or smth like that?

File: src/dexpo.cpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ main ()
               }
             if (esc)
               {
-                _exit (0); // Thread safe exit
+                return 0;
```
Comment (TheHussky) [inline src/dexpo.cpp]: What's the difference?

File: src/socket.hpp [hunk 4]
```diff
@@ -60,34 +60,43 @@ class dxp_socket
...
+      : std::runtime_error ("Got an error while reading from the socket"){};
+  explicit read_error (const std::string &msg) : std::runtime_error (msg){};
 };
 
 class bind_error : public std::runtime_error
 {
 public:
-  bind_error () : std::runtime_error ("Could not bind name to the socket"){};
+  bind_error ()
+      : std::runtime_error ("Got an error wile binding name to the socket"){};
```
Comment (TheHussky) [inline src/socket.hpp]: it is while, not wile

**Review by TheHussky — APPROVED** [review]

File: src/config.hpp (hunk 1 above)
Comment (mmskv) [reply src/config.hpp]: Fuck I accidentally pushed in my config
Will reset it in the next commit

File: src/dexpo-daemon.cpp (hunk 2 above)
Comment (mmskv) [reply src/dexpo-daemon.cpp:23]: This is standard

File: src/dexpo.cpp (hunk 3 above)
Comment (mmskv) [reply src/dexpo.cpp]: return calls destructors

**Pull request merged**